src/cfile/writer.py

```diff
@@ -515,6 +515,8 @@ def _write_function_declaration(self, elem: core.Function) -> None:
             self._write("static ")
         if isinstance(elem.return_type, core.Type):
             self._write_type_declaration(elem.return_type)
+        if isinstance(elem.return_type, core.TypeDef):
```

Comment:
Not sure if this is supposed to be here without having a unit test that proves it. Nevertheless it should say `elif` here and not `if`.

Reply:
I can chage it to elif. But should work similary. 
Well i need it here otherwise i cannot generate my code :D

Reply:
Ok this change is not necessary as i should use ``` self.cF.func_call("sizeof", [self.paramsTypes[param["data"].type]["C"]])``` instead of what i mentioned. I am developing under pressure a bit and lack of documentation is not helping :D